### ISSUE TITLE
Unhandled Promise rejection crashing app though handled

### ISSUE THREAD

**Issue by mbprodev** [twilio/TwilioChatJsReactNative#51]
I am trying to do error handling and purposefully passing an expired token into client creation method. In the below code, though I have handled the error using try -catch, my android app still crashes. The flow goes into catch block prints the log, "Inside catch" and then crashes with _Possible Unhandled Promise Rejection_
```
const Chat = require('twilio-chat'); 
....
try{
    const token=await getToken();  // returns an expired token
    const client=await Chat.Client.create(token);
    this.setState({ready:true});
}catch(err){
console.log("Inside catch");
 this.setState({ready:false});
}
```
Below is the error
 WARN  Possible Unhandled Promise Rejection (id: 0):
Error: Unable to connect: Access Token expired or expiration date invalid
construct@[native code]
_construct@http://10.0.2.2:8081/index.bundle?platform=android&dev=true&minify=false:25295:28
Wrapper@http://10.0.2.2:8081/index.bundle?platform=android&dev=true&minify=false:25239:25
TwilsockError@http://10.0.2.2:8081/index.bundle?platform=android&dev=true&minify=false:278614:82
TransportUnavailableError@http://10.0.2.2:8081/index.bundle?platform=android&dev=true&minify=false:280554:94
send@http://10.0.2.2:8081/index.bundle?platform=android&dev=true&minify=false:280282:70
http://10.0.2.2:8081/index.bundle?platform=android&dev=true&minify=false:283903:20
emit@http://10.0.2.2:8081/index.bundle?platform=android&dev=true&minify=false:269315:19
attempt@http://10.0.2.2:8081/index.bundle?platform=android&dev=true&minify=false:275850:18
_callTimer@http://10.0.2.2:8081/index.bundle?platform=android&dev=true&minify=false:31106:17
callTimers@http://10.0.2.2:8081/index.bundle?platform=android&dev=true&minify=false:31313:19
__callFunction@http://10.0.2.2:8081/index.bundle?platform=android&dev=true&minify=false:2650:49
http://10.0.2.2:8081/index.bundle?platform=android&dev=true&minify=false:2363:31
__guard@http://10.0.2.2:8081/index.bundle?platform=android&dev=true&minify=false:2604:15
callFunctionReturnFlushedQueue@http://10.0.2.2:8081/index.bundle?platform=android&dev=true&minify=false:2362:21
callFunctionReturnFlushedQueue@[native code]

![image](https://user-images.githubusercontent.com/46756021/87989816-b9e3f280-cab0-11ea-8b5d-4574a5225bc2.png)

Any suggestions on how to handle this?

**Comment by nikhilnayyar002** [comment]
Similar error on frontend. I am using reactjs not react native. Hope any solution in the thread help me also.

```
import Chat from 'twilio-chat';


    useEffect(() => {
        try {
            Chat.create(chat.token)
            .catch(err => console.log(err))
        } catch (error) {
            console.log(error)
        }
    }, [])
```



![image](https://user-images.githubusercontent.com/38173563/98685826-dcff1e80-238d-11eb-966d-4445080fbf3c.png)

**Comment by berkus** [comment]
jfyi: This all should have been fixed with JS Conversations 2.0 release.
